clone_all_repos.py

```
@@ -0,0 +1,279 @@
+    
+    print_colored(f"📥 Cloning: {repo_name}", Colors.BLUE)
+    
+    try:
+        # Use token for authentication if it's a private repo
+        if repo.get('private', False):
+            # Replace https://github.com with token-based auth
+            auth_url = clone_url.replace('https://github.com/', f'https://{os.environ.get("GITHUB_TOKEN")}@github.com/')
+        else:
+            auth_url = clone_url
+        
+        result = subprocess.run(
```

Ruff (0.13.3)

`subprocess` call: check for execution of untrusted input

(S603)